### ISSUE TITLE
Issue displaying Microsoft files (docx, xlsx...) stored on s3 buckets

### ISSUE THREAD

**Issue by lasource18** [cyntler/react-doc-viewer#228]
Hi,

Has anyone encountered this error when accessing s3 urls via DocViewerRenderers and been able to solve it?

![image](https://github.com/cyntler/react-doc-viewer/assets/68714721/957961df-8b72-4cbf-abe8-581d9515aa8e)

The scenario is, I am passing an s3 url xlsx file, once I render the document I get this error above. Here’s a snapshot of the component 

![image](https://github.com/cyntler/react-doc-viewer/assets/68714721/ca61b224-2402-4f18-b713-a8e83764db04)

I also tried to create custom renderers using Google and Microsoft viewers (see below) but I'm getting the same error

![image](https://github.com/cyntler/react-doc-viewer/assets/68714721/ece1cca9-f5cd-47d3-a16a-9e33e078b0eb)

![image](https://github.com/cyntler/react-doc-viewer/assets/68714721/ce5fdf00-553e-4172-b108-8aabe20b0e21)

![image](https://github.com/cyntler/react-doc-viewer/assets/68714721/f3730ab1-94c6-4d11-a348-162174f3287f)

The bucket policy is set as public and it works fine for pdfs and text files

I'm using MinIO to store the files and the application is built with Laravel Sail + InertiaJS (React).

If you require any additional information please let me know.

Thanks!

**Comment by rabalasubramani3** [comment]
Facing the same issue

**Comment by jiuli** [comment]
Facing the same issue

**Comment by nguyenphu123** [comment]
Facing similar issue, it shows docx or pptx files from cloudinary but when I switch to others services, It's not working.

**Comment by the-rich-piana** [comment]
Same here.

**Comment by Tsabary** [comment]
No sure if his is related to s3 buckets, Im' having issues viewing docx locally

**Comment by Akash8614** [comment]
Is this issue resolved???... Not able to load doc, docx files

**Comment by asfolif** [comment]
Same problem here. Pls fix

**Comment by ron4eg1992** [comment]
There is also a problem with viewing jgp, pdf files gives a CORS error, but docs and xlsx display normally.How fix ??help pls

**Comment by iyash1** [comment]
I'm using Pre-signed URLs from AWS S3 to get the documents. It was working before without any issues, but I have problem with MSDocs(DOC, DOCX, PPT, PPTX). Surprisingly XLX and XLSX are working fine along with the TXT, PDF, PNG and JPG files. Was anyone able to resolve the issue or have a clue what is happening?

**Comment by nampdabbank** [comment]
same issue

**Comment by gautam0923** [comment]
Same Issue

**Comment by Akshay8122** [comment]
Same for me,  weird things, XLSX and XLS are working in case of Pre-signed URLs, and DOC, DOCX, PPT, and PPTX are not working because this package claims only public URLs work in the above format. Then how are Pre-signed URLs working for XSLX and XLS and not for DOC and PPT?

**Comment by JosephScript** [comment]
I got it to work by adding the `fileType` parameter. 

```ts
const docs = [{
  uri: 'https://storage.googleapis.com/example-bucket/sample.pdf?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=redacted%40example-project.iam.gserviceaccount.com%2F20230621%2Fus-central1%2Fstorage%2Fgoog4_request&X-Goog-Date=20230621T120000Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&X-Goog-Signature=redacted',
  fileName: 'sample.pdf',
  fileType: 'pdf'
}]

return <DocViewer
  documents={docs}
  pluginRenderers={DocViewerRenderers}
/>
```

**Comment by jonathan-apptega** [comment]
> I got it to work by adding the `fileType` parameter.
> 
> const docs = [{
>   uri: 'https://storage.googleapis.com/example-bucket/sample.pdf?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=redacted%40example-project.iam.gserviceaccount.com%2F20230621%2Fus-central1%2Fstorage%2Fgoog4_request&X-Goog-Date=20230621T120000Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&X-Goog-Signature=redacted',
>   fileName: 'sample.pdf',
>   fileType: 'pdf'
> }]
> 
> return <DocViewer
>   documents={docs}
>   pluginRenderers={DocViewerRenderers}
> />

Yes, I can confirm I added the fileName and fileType to my S3 docs and it worked for xlsx and docx files. Thanks a ton @JosephScript
